src/WebJobs.Script.WebHost/ProxyFunctionExecutor.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http.Dependencies;
+using Microsoft.Azure.AppService.Proxy.Client.Contract;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Controllers;
+using Microsoft.Azure.WebJobs.Script.WebHost.WebHooks;
+
+namespace Microsoft.Azure.WebJobs.Script.Host
+{
+    public class ProxyFunctionExecutor : IFuncExecutor
+    {
+        private readonly WebScriptHostManager _scriptHostManager;
+        private readonly IDependencyResolver _dependencyResolver;
+        private FunctionsController _functionsController;
+
+        internal ProxyFunctionExecutor(WebScriptHostManager scriptHostManager, IDependencyResolver dependencyResolver, FunctionsController functionsController)
+        {
+            _scriptHostManager = scriptHostManager;
+            _dependencyResolver = dependencyResolver;
+            _functionsController = functionsController;
+        }
+
+        public async Task ExecuteFuncAsync(string funcName, Dictionary<string, object> arguments, CancellationToken cancellationToken)
```

Comment:
>arguments [](start = 87, length = 9)

What's the semantics of this dictionary? What are the keys?

Reply:
Currently the only key used is "AzureFunctionsHttpRequestKey". This allows us to pass the updated request to the FunctionExecutor.  

---
In reply to: [136211097](https://github.com/Azure/azure-webjobs-sdk-script/pull/1839#discussion_r136211097) [](ancestors = 136211097)